pkg/scheduler/plugins/topology/topology_plugin.go

```diff
@@ -32,70 +31,64 @@ func (t *topologyPlugin) Name() string {
 }
 
 func (t *topologyPlugin) OnSessionOpen(ssn *framework.Session) {
-	topologies := ssn.Topologies
-	t.nodesInfos = ssn.Nodes
-	t.initializeTopologyTree(topologies, ssn)
+	t.initializeTopologyTree(ssn.Topologies, ssn.Nodes)
 
 	ssn.AddSubsetNodesFn(t.subSetNodesFn)
 }
 
-func (t *topologyPlugin) initializeTopologyTree(topologies []*kueuev1alpha1.Topology, ssn *framework.Session) {
-	for _, singleTopology := range topologies {
-		topologyTree := &TopologyInfo{
-			Name:             singleTopology.Name,
-			DomainsByLevel:   map[string]map[TopologyDomainID]*TopologyDomainInfo{},
-			Root:             NewTopologyDomainInfo(TopologyDomainID("root"), "datacenter", "cluster", 0),
-			TopologyResource: singleTopology,
-		}
-		topologyTree.DomainsByLevel["root"] = map[TopologyDomainID]*TopologyDomainInfo{
-			topologyTree.Root.ID: topologyTree.Root,
+func (t *topologyPlugin) initializeTopologyTree(topologies []*kueuev1alpha1.Topology, nodes map[string]*node_info.NodeInfo) {
+	for _, topology := range topologies {
+		topologyTree := &Info{
+			Name: topology.Name,
+			DomainsByLevel: map[string]map[DomainID]*DomainInfo{
+				rootLevel: {
+					rootDomainId: NewDomainInfo(rootDomainId, rootLevel),
+				},
+			},
+			TopologyResource: topology,
 		}
 
-		for _, nodeInfo := range ssn.Nodes {
-			t.addNodeDataToTopology(topologyTree, singleTopology, nodeInfo)
+		for _, nodeInfo := range nodes {
+			t.addNodeDataToTopology(topologyTree, topology, nodeInfo)
 		}
 
-		t.TopologyTrees[singleTopology.Name] = topologyTree
+		t.TopologyTrees[topology.Name] = topologyTree
 	}
 }
 
-func (*topologyPlugin) addNodeDataToTopology(topologyTree *TopologyInfo, singleTopology *kueuev1alpha1.Topology, nodeInfo *node_info.NodeInfo) {
+func (*topologyPlugin) addNodeDataToTopology(topologyTree *Info, topology *kueuev1alpha1.Topology, nodeInfo *node_info.NodeInfo) {
 	// Validate that the node is part of the topology
-	if !isNodePartOfTopology(nodeInfo, singleTopology) {
+	if !isNodePartOfTopology(nodeInfo, topology) {
 		return
 	}
 
-	var nodeContainingChildDomain *TopologyDomainInfo
-	for levelIndex := len(singleTopology.Spec.Levels) - 1; levelIndex >= 0; levelIndex-- {
-		level := singleTopology.Spec.Levels[levelIndex]
+	var nodeContainingChildDomain *DomainInfo
+	for levelIndex := len(topology.Spec.Levels) - 1; levelIndex >= 0; levelIndex-- {
+		level := topology.Spec.Levels[levelIndex]
 
-		domainName, foundLevelLabel := nodeInfo.Node.Labels[level.NodeLabel]
-		if !foundLevelLabel {
-			continue // Skip if the node is not part of this level
-		}
```

Comment:
not needed?

Reply:
Is it the same? in `isNodePartOfTopology` it checks that node is part of some topology. here it skips specific levels.